core/test/unit/helpers/t_spec.js

```diff
@@ -0,0 +1,30 @@
+// Test: {{t}} translation helper
+// Compatibility with both old themes and i18n-capable themes
+
+var should = require('should'),
+    helpers = require('../../../server/helpers'),
+    proxy = require('../../../server/helpers/proxy'),
+    i18n = proxy.i18n;
+
+describe('{{t}} helper', function () {
+    var locale = i18n.locale();
+    if (locale.substring(0, 2) === 'en') {
+        it('returns correct frontend text string', function () {
```

Comment:
What is the difference between both tests? 🤔 
I think the second case can be removed, they look like they are doing the same.
If it was added during developing, that's fine. But no need to keep.

Reply:
The first one was for English, the second was for translation to another language, depending on the locale set for the tested installation. That was testing in May.

Reply:
Please no conditional tests. Either you mock `settingsCache.get(..)` or you remove the test 👍

Reply:
Testing with `grunt validate`, it's always giving the same, no matter if the locale setting in database is English or another one:
```
{{t}} helper
    ✓ returns correct frontend text string
```
That should be the result for English, so it looks like the tests don't use the settings in the database. Probably it's falling back to the English default.

And, if using the default Casper theme without language files, it's falling back to the default string passed by the translation helper. So it's just testing that applying the helper to a non-translatable theme gives the default string.